### PR TITLE
Removing cirq_google.optimizers

### PR DESCRIPTION
**Pull request by verult**
@tanujkhattar

### CODE REVIEW

File: cirq-google/cirq_google/optimizers/two_qubit_gates/example.py [hunk 1]
```diff
@@ -1,104 +0,0 @@
-# pylint: disable=wrong-or-nonexistent-copyright-notice
-"""Example application of the two-qubit gate compilation algorithm.
```
Comment (tanujkhattar) [inline cirq-google/cirq_google/optimizers/two_qubit_gates/example.py:2]: I think we should probably keep this example around; there's nothing deprecated in here. Not sure what's the best place to move this to; Maybe Cirq/examples?

**Review by tanujkhattar — DISMISSED** [review]
This is lovely! Just one comment about moving the gate tabulation example to Cirq/examples and cleaning it up a little (eg: add copyright notice on top; etc.)

File: cirq-google/cirq_google/optimizers/two_qubit_gates/example.py (hunk 1 above)
Comment (verult) [reply cirq-google/cirq_google/optimizers/two_qubit_gates/example.py]: Thanks for the catch!

**Comment by verult** [comment]
Removed the directory from `cirq_google/__init__.py`, and also removed `GateTabulation` JSON data

File: cirq-google/cirq_google/json_resolver_cache.py [hunk 2]
```diff
@@ -44,7 +44,6 @@ def _old_xmon(*args, **kwargs):
         'CouplerPulse': cirq_google.experimental.CouplerPulse,
         'GoogleNoiseProperties': cirq_google.GoogleNoiseProperties,
         'SycamoreGate': cirq_google.SycamoreGate,
-        'GateTabulation': cirq_google.GateTabulation,
```
Comment (dstrain115) [inline cirq-google/cirq_google/json_resolver_cache.py:47]: We cannot delete these json files.  We need to provide a way for people to deserialize this:

See
https://quantumai.google/cirq/dev/serialization#deprecating_a_serializable_value

An example where I recently did this was here:
https://github.com/quantumlib/Cirq/pull/5687

**Review by dstrain115 — COMMENTED** [review]
We cannot delete the json files.

Also, seems to be a deprecation check in the quantum volume test that needs to be modified to fix the tests.

**Comment by verult** [comment]
Fixed json deserialization and quantum_volume. By inspection, the quantum_volume example doesn't seem to expect compiled circuits of a particular form, so `GoogleCZTargetGateset` won't be necessary.

File: examples/advanced/quantum_volume_test.py [hunk 3]
```diff
@@ -11,10 +11,7 @@ def test_main_loop():
     # Keep test from taking a long time by lowering repetitions.
     pytest.importorskip("cirq_google")
     args = '--num_qubits 5 --depth 5 --num_circuits 1  --routes 3'.split()
-    with cirq.testing.assert_deprecated(
-        'Use cirq.optimize_for_target_gateset', deadline='v0.16', count=None
-    ):
-        quantum_volume.main(**quantum_volume.parse_arguments(args))
+    quantum_volume.main(**quantum_volume.parse_arguments(args))
```
Comment (dstrain115) [inline examples/advanced/quantum_volume_test.py:14]: Looks like the 'import cirq' is now unneeded (and causing the pylint error.

**Review by dstrain115 — APPROVED** [review]
LGTM aside from the tests failures.

Looks like advanced/quantum_volume_errors.ipynb and advanced/quantum_volume_routing.ipynb also have references to optimized_for_xmon.

It would be nice to get one more person to take a quick look at the json backwards compat thing (if possible under the time constraints) just to double-check, but I am pretty sure this is the correct way to do it.

**Comment by verult** [comment]
cc @95-martin-orion in case you have time to take a look at the JSON backward compatibility solution. Thanks!

File: cirq-google/cirq_google/json_test_data/GateTabulation.json [hunk 4]
```diff
@@ -1,519 +0,0 @@
-{
```
Comment (tanujkhattar) [inline cirq-google/cirq_google/json_test_data/GateTabulation.json:1]: We should have moved the `GateTabulation.json` to `GateTabulation.json_inward`; which matches the repr in `GateTabulation.repr_inward` ? I see the `GateTabulation.repr_inward` file but I don't see the `GateTabulation.json_inward` file.

**Review by tanujkhattar — COMMENTED** [review]
One question about json changes; LGTM otherwise.

File: cirq-google/cirq_google/json_test_data/GateTabulation.json (hunk 4 above)
Comment (verult) [reply cirq-google/cirq_google/json_test_data/GateTabulation.json:1]: Shoot thanks for the catch. It's in my local repo but wasn't added.

Comment (verult) [reply cirq-google/cirq_google/json_test_data/GateTabulation.json:1]: Actually what happened was that `GateTabulation.json_inward` was already there and unmodified. Same with `GateTabulation.repr_inward`, but the class name was modified to be `cirq.TwoQubitGateTabulation`. So we are all good on that front.

**Pull request merged**